src/libsyntax/ext/expand.rs

```diff
@@ -390,30 +388,91 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
         }
     }
 
-    fn collect_invocations(&mut self, expansion: Expansion, derives: &[Mark])
-                           -> (Expansion, Vec<Invocation>) {
-        let result = {
+    fn collect_invocations(&mut self,
```

Comment:
@abonander 
Did you figure out what happens here and above in `fn expand`?
I can't understand it just by reading the code, but it usually becomes more clear after actively doing some refactoring or modifications.

Reply:
I'm not sure what you mean by "figure out what happens here". It seems to do what it says on the tin; it accumulates bang/attribute macro invocations and then `expand` expands them. It has to be done in a loop because macro expansions can produce more invocations.